theblues/plans.py

```diff
@@ -0,0 +1,50 @@
+from collections import namedtuple
+import datetime
+
+from theblues.errors import (
+    log,
+    ServerError,
+)
+from theblues.utils import (
+    ensure_trailing_slash,
+    make_request,
+)
+
+Plan = namedtuple('Plan', 'url plan created_on description price')
+
+
+class Plans(object):
+
+    def __init__(self, url):
+        """Initializer.
+
+        @param url The url to the Plan API.
+        """
+        self.url = ensure_trailing_slash(url)
+
+    def get_plans(self, reference):
+        """ Get the plans for a given charm.
+
+        @param the Reference to a charm
+        @return a list of plans or an empty tuple if no plans.
+        @raise ServerError
+        """
+        json = make_request(
+            '{}charm?charm-url={}'.format(self.url,
+                                          'cs:' + reference.path()))
```

Comment:
IIRC the full path including cs schema can be generated by just doing `reference.id()`.